cpp/daal/src/data_management/finiteness_checker_avx2_impl.i

```diff
@@ -0,0 +1,234 @@
+    float sum;
+
+    __m256 sums     = _mm256_set1_ps(0);
+    __m256 * ptr256 = (__m256 *)dataPtr;
+    for (size_t i = 0; i < n / nPerInstr; i++) sums = _mm256_add_ps(sums, ptr256[i]);
+
+    // AVX2 doesn't have reduce_add_ps, so finer-grained used of intrinsics necessary
+    __m128 s0    = _mm256_castps256_ps128(sums);
+    __m128 s1    = _mm256_extractf128_ps(sums, 1);
+    s0           = _mm_add_ps(s0, s1);
+    __m128 stemp = _mm_movehl_ps(s0, s0);
+    __m128 sD    = _mm_add_ps(s0, stemp);
+    __m128 hi    = _mm_shuffle_ps(sD, sD, 0x1);
+
+    sum = _mm_cvtss_f32(_mm_add_ss(sD, hi));
+
+    for (size_t i = (n / nPerInstr) * nPerInstr; i < n; ++i) sum += dataPtr[i];
+
+    return sum;
+}
+
+template <>
+double sumWithAVX<double, avx2>(size_t n, const double * dataPtr)
+{
+    constexpr size_t avx2RegisterLength = 256;
+    constexpr size_t numberOfBitsInByte = 8;
+    constexpr size_t nPerInstr          = avx2RegisterLength / (numberOfBitsInByte * sizeof(double));
+    double sum;
+
+    __m256d sums     = _mm256_set1_pd(0);
+    __m256d * ptr256 = (__m256d *)dataPtr;
+    for (size_t i = 0; i < n / nPerInstr; i++) sums = _mm256_add_pd(sums, ptr256[i]);
+
+    // AVX2 doesn't have reduce_add_pd, so finer-grained used of intrinsics necessary
+    __m128d s0    = _mm256_castpd256_pd128(sums);
+    __m128d s1    = _mm256_extractf128_pd(sums, 1);
+    s0            = _mm_add_pd(s0, s1);
+    __m128d stemp = _mm_unpackhi_pd(s0, s0);
+
+    sum = _mm_cvtsd_f64(_mm_add_sd(s0, stemp));
+
+    for (size_t i = (n / nPerInstr) * nPerInstr; i < n; ++i) sum += dataPtr[i];
+
+    return sum;
+}
+
+template <>
+float computeSum<float, avx2>(size_t nDataPtrs, size_t nElementsPerPtr, const float ** dataPtrs)
+{
+    // computeSumAVX defined in finiteness_checker_cpu.cpp
+    return computeSumAVX<float, avx2>(nDataPtrs, nElementsPerPtr, dataPtrs);
+}
+
+template <>
+double computeSum<double, avx2>(size_t nDataPtrs, size_t nElementsPerPtr, const double ** dataPtrs)
+{
+    // computeSumAVX defined in finiteness_checker_cpu.cpp
+    return computeSumAVX<double, avx2>(nDataPtrs, nElementsPerPtr, dataPtrs);
+}
+
+template <>
+double computeSumSOA<avx2>(NumericTable & table, bool & sumIsFinite, services::Status & st)
+{
+    // computeSumSOAAVX defined in finiteness_checker_cpu.cpp
+    return computeSumSOAAVX<avx2>(table, sumIsFinite, st);
+}
+
+template <>
+services::Status checkFinitenessInBlocks<avx2>(const float ** dataPtrs, bool inParallel, size_t nTotalBlocks, size_t nBlocksPerPtr, size_t nPerBlock,
+                                               size_t nSurplus, bool allowNaN, bool & finiteness)
+{
+    services::Status s;
+    constexpr size_t avx2RegisterLength = 256;
+    constexpr size_t numberOfBitsInByte = 8;
+    constexpr size_t nPerInstr          = avx2RegisterLength / (numberOfBitsInByte * sizeof(float));
+    services::internal::TArray<bool, avx2> notFiniteArr(nTotalBlocks);
+    bool * notFinitePtr = notFiniteArr.get();
```

Comment:
This is just a FYI. No action is needed.

It might be better (in terms of performance) to have those kind of arrays as a 'thread local storage' (see daal::tls).

Having them as a regular arrays might lead to cache coherency issues when i-th thread updates i-th element of notFinitePtr array, and (i+1)-th thread updates (i+1)-th element of the same array. In this case both threads might need to synchronize the information in their caches as both i-th and (i+1)-th elements of the array might reside in both caches of i-th and (i+1)-th threads and can become unsynchronized after the update.